lib/game_play.rb

```diff
@@ -59,25 +60,37 @@ def create_players(selection)
     end
   end
 
+  def get_user_selection_between(min, max)
+    value = get_user_selection
+    if !value.between?(min, max)
+      @ui.between(min, max)
+      value = get_user_selection_between(min, max)
+    end
+
+    value
+  end
+
   def get_user_selection
     begin
-      Integer(@ui.get_value)
+      selection = Integer(@ui.get_value)
     rescue
       @ui.must_be_integer
-      get_user_selection
+      selection = get_user_selection
     end
+
+    selection
   end
 
-  def creation_type_game(type_game)
-    if type_game == 1
-      @player_one = Human.new("X", ui)
-      @player_two = Human.new("O", ui)
-    elsif type_game == 2
-      @player_one = Human.new("X", ui)
-      @player_two = Computer.new("O", ui, board)
-    elsif type_game == 3
-      @player_one = Computer.new("X", ui, board)
-      @player_two = Computer.new("O", ui, board)
+  def creation_type_game(type_selected)
+    if type_selected == GameType::HUMAN_VS_HUMAN
```

Comment:
If I asked for a new player type to be introduced, you would need to change this file, even though the rest of the game rules would stay the same. For this reason, I would suggest that this method is actually a single responsibility which could be put in it's own class.